front-end-tools/src/Main.tsx

```diff
@@ -0,0 +1,579 @@
+                        <i>Loading connector...</i>
+                    </p>
+                )}
+                {connectError && (
+                    <p className="alert alert-danger" role="alert">
+                        Connect Error: {connectError}.
+                    </p>
+                )}
+                {!connection && !isWaitingForTransaction && activeConnectorType && activeConnector && (
+                    <p>
+                        <button className="btn btn-primary me-1" type="button" onClick={connect}>
+                            {isConnecting && 'Connecting...'}
+                            {!isConnecting && activeConnectorType === BROWSER_WALLET && 'Connect Browser Wallet'}
+                        </button>
+                    </p>
+                )}
+                {connection && !accountExistsOnNetwork && (
+                    <>
+                        <div className="alert alert-danger" role="alert">
+                            Please ensure that your browser wallet is connected to network `
+                            {walletConnectionProps.network.name}` and you have an account in that wallet that is
+                            connected to this website.
+                        </div>
+                        <div className="alert alert-danger" role="alert">
+                            Alternatively, if you intend to use `{isTestnet ? 'mainnet' : 'testnet'}`, switch the
+                            network button at the top of this webpage.
+                        </div>
+                    </>
+                )}
+            </div>
+            {account && (
+                <div className="row">
+                    {connection && account !== undefined && (
+                        <div className="col-lg-12">
+                            {viewError && (
+                                <div className="alert alert-danger" role="alert">
+                                    Error: {viewError}.
+                                </div>
+                            )}
+                            <br />
+                            <div className="label">Connected account:</div>
+                            <div>
+                                <a
+                                    className="link"
+                                    href={`https://${
+                                        isTestnet ? `testnet.` : ``
+                                    }ccdscan.io/?dcount=1&dentity=account&daddress=${account}`}
+                                    target="_blank"
+                                    rel="noreferrer"
+                                >
+                                    {account}
+                                </a>
+                            </div>
+                            <br />
+                            <div className="label">Your account balance:</div>
+                            <div>{accountBalance.replace(/(\d)(?=(\d\d\d\d\d\d)+(?!\d))/g, '$1.')} CCD</div>
```

Comment:
![Screenshot from 2023-06-22 09-01-05](https://github.com/Concordium/concordium-smart-contract-tools/assets/105212/875439f7-d1eb-4a01-a906-e1745129414d)

It would be ideal if I was not allowed to try to deploy the module if I don't have enough balance. But there might not be good helpers in the web SDK to allow you to precompute this easily.

Also, this looks like a hacky way of displaying CCD amounts with a `.`. Is there no library function in the Web SDK to display amounts in CCD? If there isn't it should be added.

Reply:
https://github.com/Concordium/concordium-node-sdk-js/issues/215
It would be good to have a simulate transaction/feature in the webSDK, so a front-end can first call to simulate a specific transaction to see if it would revert (e.g. because insufficient funds in the wallet or because of a contract revert) and only if the simulation was successful, allow the user the execution of the real transaction on the front end. It would be similar to the invokeContract call (which can simulate Transaction.Update transactions). This request is to expand the simulation to any type of transaction e.g. deploy/initilize transactions as well.

This display of CCD is something that should be done in the browser, e.g. get the browser setting and based on your language/region, you either use a '.' or a ',' after the full CCD amount. I feel that is not something the Web SDK can do. 

What the web SDK could do is convert `1 CCD` into 1000000 microCCD which would be nice.

Reply:
As we discussed in the planning meeting, this needs some node support first, which we will add in Q3. After that is doen we should definitely have this.